torch/_inductor/select_algorithm.py

```diff
@@ -609,6 +609,7 @@ def autotune(choice):
             choices,
             choices[0].name,
             repr([self.key_of(x) for x in input_nodes]),
+            repr(layout),
```

Comment:
Add an:
```
assert isinsance(layout, ir.FixedLayout)
```
Want to make sure we aren't getting a FlexibleLayout here.

Reply:
Just for clarification, is it the case that we shouldn't be getting ir.FlexibleLayout here or that we just shouldn't be including it in that case?

Reply:
If we are, it is a bug.

Reply:
Ok I'll have to look into that, since we are getting flexible layouts here.

Reply:
You should just call `.freeze_layout()` to convert flexible->fixed.